src/garage/_dtypes.py

```diff
@@ -473,6 +473,35 @@ class StepType(enum.IntEnum):
     # Denotes the last `TimeStep` in a sequence truncated by time limit.
     TIMEOUT = 3
 
+    @classmethod
+    def get_step_type(cls, step_cnt, max_episode_length, done):
```

Comment:
why not make this a method of StepType instead?

Reply:
Hmm what do you mean? Don't make it as a classmethod?

Reply:
no -- i mean literally to have it as `StepType.which()` or something --as a method of StepType

Reply:
Just curious: why do we want to make this change and why is this better?

Reply:
because it would be useful to any user of StepType, not just users of TrajectoryBatch

Reply:
wait but this is a function for StepType, not TrajectoryBatch

Reply:
sorry i misread the giant diff :)